src/SnowAgingSnicarMod.F90

```diff
@@ -0,0 +1,273 @@
+! There are two parameters that describe the grain growth rate as 
+! a function of snow liquid water content (LWC). The "LWC=0" parameter
+! is zeroed here because we are accounting for dry snowing with a 
+! different representation
+!
+! Re-freezing of liquid water: Assume that re-frozen liquid water clumps
+! into an arbitrarily large effective grain size (SnowRadiusRefrz). 
+! The phenomenon is observed (Grenfell), but so far unquantified, as far as 
+! I am aware.
+! -------------------------------------------------------------------------
+
+    implicit none
+
+    type(noahmp_type), intent(inout) :: noahmp
+
+! local variable
+    integer                          :: SnowLayerTop                         ! top snow layer index [idx]
+    integer                          :: SnowLayerBottom                      ! bottom snow layer index [idx]
+    integer                          :: TemperatureInd                       ! snow aging lookup table temperature index [idx]
+    integer                          :: TemperatureGradientInd               ! snow aging lookup table temperature gradient index [idx]
+    integer                          :: SnowDensityInd                       ! snow aging lookup table snow density index [idx]
+    integer                          :: LoopInd                              ! do loop/array indices
+    integer,  parameter              :: idx_T_min         = 1                ! minimum temperature index used in aging lookup table [idx]
+    integer,  parameter              :: idx_Tgrd_min      = 1                ! minimum temperature gradient index used in aging lookup table [idx]
+    integer,  parameter              :: idx_rhos_min      = 1                ! minimum snow density index used in aging lookup table [idx]
+    real(kind=kind_noahmp)           :: bst_tau                              ! best fit snow aging parameter retrieved from lookup table [hour]
+    real(kind=kind_noahmp)           :: bst_kappa                            ! best fit snow aging parameter retrieved from lookup table [unitless]
+    real(kind=kind_noahmp)           :: bst_drdt0                            ! best fit snow aging parameter retrieved from lookup table [um hr-1]
+    real(kind=kind_noahmp)           :: h2osno_lyr                           ! liquid + solid H2O in snow layer [kg m-2]
+    real(kind=kind_noahmp)           :: TemperatureSnowLayerTop              ! temperature at upper layer boundary [K]
+    real(kind=kind_noahmp)           :: TemperatureSnowLayerBottom           ! temperature at lower layer boundary [K]
+    real(kind=kind_noahmp)           :: SnowDensity                          ! snow density [kg m-3]
+    real(kind=kind_noahmp)           :: SnowRadiusMax     = 1500             ! maximum allowed snow effective radius [microns]
+    real(kind=kind_noahmp), parameter:: C1_liq_Brun89     = 0.0              ! constant for liquid water grain growth [m3 s-1], from Brun89: zeroed to accomodate dry snow aging
```

Comment:
@tslin2 any specific reason to turn off the wet aging process? What value is used in CLM?

Reply:
ok, I saw the justification from CLM code. this is fine